### PR TITLE
IA 734 polio email notification

### PR DESCRIPTION
**Pull request by quang-le**
## Features

- Added a "Configuration" entry to the polio menu
- Display a Table with countries, users to notify, selected language and an "Edit" icon
- Clicking Edit opens a modal with 2 select boxes allowing to modify users' list and language
- Users can sort by country or by language
- Default sorting is by country
- Only users with an email address can be chosen from dropdown list
- Created `TableWithDeepLink` to simplify management of deep linking in `Table`

## Changes
- Added possibility to choose the size of `IconButton` (bluesquare-components)
- Added `onTableParamsChange` prop to `Table` (bluesquare-components)
- Updated dependency to bluesquare-components
- Fixed a typo here and there

https://user-images.githubusercontent.com/38907762/131904697-85fdb3c4-512e-439a-b7fe-2d44193ec371.mov

### CODE REVIEW

File: hat/assets/js/apps/Iaso/components/tables/SingleTable.js [hunk 1]
```diff
@@ -56,10 +56,9 @@ const SingleTable = ({
     const [loading, setLoading] = useState(false);
     const [selection, setSelection] = useState(selectionInitialState);
     const [didFetchData, setDidFetchData] = useState(false);
-    const [firstLoad, setfFrstLoad] = useState(true);
+    const [firstLoad, setfFirstLoad] = useState(true);
```
Comment (olethanh) [inline hat/assets/js/apps/Iaso/components/tables/SingleTable.js:59]: it was also bugging me but I think it should be `setFirstLoad` with one less `f`

File: hat/assets/js/apps/Iaso/components/tables/TableWithDeepLink.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import { Table } from 'bluesquare-components';
+import { string } from 'prop-types';
+import { handleTableDeepLink } from '../../utils/table';
+
+export const TableWithDeepLink = ({ baseUrl, ...props }) => {
+    const onTableParamsChange = handleTableDeepLink(baseUrl);
+    const newProps = { ...props };
+    newProps.onTableParamsChange = onTableParamsChange;
+    return <Table {...newProps} />;
```
Comment (olethanh) [inline hat/assets/js/apps/Iaso/components/tables/TableWithDeepLink.js:10]: you can directly do:
```<Table {...newProps} onTableParamsChange={handleTableDeepLink(baseUrl)} />```

File: plugins/polio/js/src/components/EmailNotifications/EmailNotificationsConfig.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import { withRouter } from 'react-router';
+import { Box, makeStyles } from '@material-ui/core';
+import { commonStyles } from 'bluesquare-components';
+import TopBar from '../../../../../../hat/assets/js/apps/Iaso/components/nav/TopBarComponent';
+import { EmailNotificationsTable } from './Table/EmailNotificationsTable';
+
+const useStyles = makeStyles(theme => ({
+    ...commonStyles(theme),
+}));
+const EmailNotificationConfig = ({ params }) => {
```
Comment (olethanh) [inline plugins/polio/js/src/components/EmailNotifications/EmailNotificationsConfig.js:12]: please rename to countryConfig and as well as the  others Component

**Review by beygorghor — APPROVED** [review]

File: hat/assets/js/apps/Iaso/components/tables/TableWithDeepLink.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import { Table } from 'bluesquare-components';
+import { string } from 'prop-types';
+import { handleTableDeepLink } from '../../utils/table';
+
+export const TableWithDeepLink = ({ baseUrl, ...props }) => {
+    const onTableParamsChange = handleTableDeepLink(baseUrl);
+    const newProps = { ...props };
+    newProps.onTableParamsChange = onTableParamsChange;
```
Comment (olethanh) [inline hat/assets/js/apps/Iaso/components/tables/TableWithDeepLink.js:9]: to remove

**Review by olethanh — APPROVED** [review]

**Pull request merged**